src/cache.ts

```diff
@@ -98,17 +131,38 @@ export async function getBinary(
       );
     })[0];
 
-    console.log(
-      "Downloading browser... (this may take a while depending on your internet connection)",
-    );
     const req = await fetch(download.url);
     if (!req.body) {
       throw new Error(
         "Download failed, please check your internet connection and try again",
       );
     }
-    await Deno.writeFile(resolve(BASE_PATH, `raw_${VERSION}.zip`), req.body);
-    console.log(`Download complete (${browser} version ${VERSION})`);
+    const reader = req.body.getReader();
+    const archive = await Deno.open(resolve(BASE_PATH, `raw_${VERSION}.zip`), {
+      write: true,
+      truncate: true,
+      create: true,
+    });
+    const bar = new ProgressBar({
+      title: `Downloading ${browser} ${VERSION}`,
+      total: Number(req.headers.get("Content-Length") ?? 0),
+      clear: true,
+      display: ":title :bar :percent",
+    });
+    let downloaded = 0;
+    do {
+      const { done, value } = await reader.read();
+      if (done) {
+        break;
+      }
+      await Deno.write(archive.rid, value);
+      downloaded += value.length;
+      bar.render(downloaded);
+    } while (true);
+    Deno.close(archive.rid);
+    if (!await isQuietInstall()) {
+      console.log(`Download complete (${browser} version ${VERSION})`);
+    }
```

Comment:
Is the performance of this about the same as `writeFile`? I'm a little concerned that this might be too slow.

Reply:
Normally it should be around the same yes (deno use something similar internally when it cannot use the op_fs directly: https://github.com/denoland/deno/blob/29026fac21d85a530d87ca3e94ae0d547557fa23/ext/fs/30_fs.js#L779-L804)
Since it's based on the [`ReadableStream`](https://developer.mozilla.org/en-US/docs/Web/API/ReadableStream/getReader) and [`WritableStream`](https://developer.mozilla.org/en-US/docs/Web/API/WritableStream), there shouldn't be too much as the operation is similar to a `pipeTo()`, it's just that it peak at the chunk length while the stream is read but since it doesn't copy data into intermediate buffer I don't think it impact performances that much

But maybe for the quiet install we could optimize and fallback on the previous way of writing?

Reply:
Sure. I think quiet install going back to the other path makes sense to me.

Reply:
![image](https://github.com/lino-levan/astral/assets/22963968/7cac90dc-fc05-451a-af34-943bad2b3ca5)

I made a small benchmark just to check if it was correct, seems that there isn't a big difference.
I feel like it's worth it to have the progress so you know the programm isn't stuck, but lmk if you'd prefer this to not be the default behaviour

Reply:
I think this is valuable. Thanks for the work. Let me review dependencies and I think this PR looks good. I'll give it one more lookover.